src/stories/common/ButtonWithIcon.stories.js

```diff
@@ -0,0 +1,57 @@
+import ButtonWithIcon from "../../components/common/ButtonWithIcon";
+import {faStar} from "@fortawesome/free-solid-svg-icons";
+
+export default {
+    title: 'Common/ButtonWithIcon',
+    component: ButtonWithIcon,
+    argTypes: {
+        type: {
+            control: { type: 'select' },
+            options: ['primary', 'secondary', 'success', 'danger']
+        },
+        onClick: { action: 'clicked' },
+    },
+};
+
+const Template = (args) => ({
+    components: { ButtonWithIcon },
+    setup() {
+        return { args };
+    },
+    template: '<button-with-icon v-bind="args">Test Button</button-with-icon>',
+});
+
+export const Primary = Template.bind({});
```

Comment:
You wont need to make every version of the button in the story. Just a primary button will be enough.
Unless you make a change that cannot be previewed yet in the normal base-button.